lib/Registrator.js

```diff
@@ -165,9 +169,25 @@ ${this._contact};expires=${this._expires}${this._extraContactParams}`);
               .reduce((a, b) => a.concat(b.parsed), []);
 
             // Get the Contact pointing to us and update the expires value accordingly.
-            const contact = contacts.find((element) => (
-              element.uri.user === this._ua.contact.uri.user
-            ));
+            let contact = undefined;
+
+            /* If we have a reg-id and a sip.instance parameter in our Contact header then try to
+               find a matching Contact using that. */
+            if (this._reg_id && this._sipInstance)
```

Comment:
This condition is not needed because both members are always set.

Reply:
I just had those checks there in case someone changed the initialization in the constructor, but I agree, they are not really needed.